src/Analysis/Codelyzer.Analysis.Common/FileUtils.cs

```diff
@@ -39,5 +33,40 @@ public static string GetRelativePath(string filePath, string dirPath)
             var path = filePath.Replace(dirPathSeparator, "");
             return path;
         }
+
+        public static void DirectoryCopy(string sourceDirName, string destDirName, bool copySubDirs = true)
```

Comment:
Another question on public API surfaces: Is it okay to put this here? It seems like a more robust solution (mostly copied from Microsoft documentation). I saw CTA is doing something similar on it's own. Should CTA be using this instead once/if it is published? CTA: https://github.com/aws/cta/blob/dcdf4dde2c6a105b0fe2115c3f275686999806b1/src/CTA.Rules.Config/Utils.cs#L216

Reply:
Since CTA ports code and Codelyzer analyzes it, I have no strong opinion on where FileUtils should live. If we do end up sharing code, I agree it should be in Codelyzer since it's the lowest level dependency.

Reply:
We may end up creating a support library project but as things currently stand I agree with John the lowest level dependency being Codelyzer should hold this for now.